src/sentry/tagstore/snuba/backend.py

```diff
@@ -860,6 +861,36 @@ def _get_tag_values_for_semver_build(self, projects, environments, build):
             [(i, TagValue(SEMVER_BUILD_ALIAS, v, None, None, None)) for i, v in enumerate(packages)]
         )
 
+    def _get_tag_values_for_releases_across_all_datasets(self, projects, environments, query):
+        from sentry.api.paginator import SequencePaginator
+
+        organization_id = Project.objects.filter(id=projects[0]).values_list(
+            "organization_id", flat=True
+        )[0]
+        qs = Release.objects.filter(organization_id=organization_id)
+
+        if projects:
+            qs = qs.filter(
+                id__in=ReleaseProject.objects.filter(project_id__in=projects).values_list(
+                    "release_id", flat=True
+                )
+            )
+        if environments:
+            qs = qs.filter(
+                id__in=ReleaseEnvironment.objects.filter(
+                    environment_id__in=environments
+                ).values_list("release_id", flat=True)
+            )
+
+        if query:
+            qs = qs.filter(version__startswith=query)
+
+        versions = qs.order_by("version").values_list("version", flat=True)[:1000]
```

Comment:
We should probably have a distinct here, otherwise I think we can get duplicate releases if they exist on multiple projects/envs. Probably worth writing a test to verify

Reply:
How could that happen? We are not doing joins anywhere

Reply:
Good point, lgtm then